### PR TITLE
wallet error & tor crash fix

### PR DESCRIPTION
**Pull request by SerhiiShovkoplias**
added timer for wait while wallet restart
added check if TorThread in progress for start new

## Description
added timer for wait while wallet restart
added check if TorThread in progress for start new

## Motivation and Context
* [x] tari-project/wallet-ios#339
* [x] tari-project/wallet-ios#364
* [x] tari-project/wallet-ios#429

## How Has This Been Tested?
run tests, manual testing

## Types of changes

* [ ] UI fix (non-breaking change which fixes a UI issue)
* [x] Functionality bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I have tested this on multiple simulator devices with different screensizes.
* [x] I'm merging against the `development` branch.
* [x] Commits have been squashed into one commit with a descriptive commit message
* [x] I ran all tests before pushing.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added/changed tests to cover my changes if not UI changes.

### CODE REVIEW

File: MobileWallet/TariLib/Tor/OnionManager.swift [hunk 1]
```diff
@@ -195,53 +195,12 @@ public class OnionManager: NSObject {
...
             }
         }
-
         // Wait long enough for tor itself to have started. It's OK to wait for this
         // because Tor is already trying to connect; this is just the part that polls for
         // progress.
         DispatchQueue.main.asyncAfter(deadline: .now() + 1, execute: { [weak self] in
             guard let self = self else { return }
 
-//            if OnionManager.isTorLogging {
```
Comment (Jasonvdb) [inline MobileWallet/TariLib/Tor/OnionManager.swift:205]: This code is actually something I want to keep trying to add. It fails on travis but I'm checking it every time the tor lib is updated.

File: MobileWallet/TariLib/Tor/OnionManager.swift [hunk 2]
```diff
@@ -170,7 +170,7 @@ public class OnionManager: NSObject {
             self.torController = TorController(socketHost: OnionManager.CONTROL_ADDRESS, port: OnionManager.CONTROL_PORT)
         }
 
-        if ((self.torThread == nil) || (self.torThread?.isCancelled ?? true)) {
+        if (self.torThread == nil || self.torThread?.isCancelled == true) && TorThread.active?.isExecuting != true {
```
Comment (Jasonvdb) [inline MobileWallet/TariLib/Tor/OnionManager.swift:173]: What does this achieve? Or what is it fixing?

Comment (SerhiiShovkoplias) [reply MobileWallet/TariLib/Tor/OnionManager.swift:173]: when we call thread.cancel() we set only flag but thread still going. This expression checks if thread has tasks.

File: MobileWallet/TariLib/Tor/OnionManager.swift (hunk 1 above)
Comment (SerhiiShovkoplias) [reply MobileWallet/TariLib/Tor/OnionManager.swift]: `DispatchQueue.main.asyncAfter(deadline: .now() + 1,...` does not guarantee that active TorThread finish all tasks.

File: MobileWallet/TariLib/Tor/OnionManager.swift (hunk 2 above)
Comment (Jasonvdb) [reply MobileWallet/TariLib/Tor/OnionManager.swift]: Ok cool, were you able to replicate the issue and then test that this fixes it every time?

Comment (SerhiiShovkoplias) [reply MobileWallet/TariLib/Tor/OnionManager.swift]: @Jasonvdb  When I deleted this code, I managed to play this crash. Then I brought it back and the crash was never reproduced again.

**Pull request merged**